### DISCUSSION TITLE
NFS 4.1 Support

### DISCUSSION THREAD

**Discussion by FaraSys** [longhorn/longhorn#5152] [Q&A]
Hello. Is NFS 4.1 support planned in the roadmap?

**Comment by derekbit** [comment]
For what?

**Reply by FaraSys** [reply]
I am using Windows server 2019 with nfs v4.1 as NFS server. When I try to connect nfs to Longhorn I get an error about an unsupported version of nfs.
`error listing backup volume names: failed to execute: /var/lib/longhorn/engine-binaries/longhornio-longhorn-engine-v1.3.2/longhorn [backup ls --volume-only nfs://10.128.61.162:/longhornTest/], output Cannot mount nfs 10.128.61.162:/longhornTest/: vers=4.0: Failed to execute: mount [-t nfs4 -o nfsvers=4.0 -o actimeo=1 10.128.61.162:/longhornTest/ /var/lib/longhorn-backupstore-mounts/10_128_61_162/longhornTest], output mount.nfs4: Protocol not supported , error exit status 32: vers=4.1: Failed to execute: mount [-t nfs4 -o nfsvers=4.1 -o actimeo=1 10.128.61.162:/longhornTest/ /var/lib/longhorn-backupstore-mounts/10_128_61_162/longhornTest], output mount.nfs4: mounting 10.128.61.162:/longhornTest/ failed, reason given by server: No such file or directory , error exit status 32: vers=4.2: Failed to execute: mount [-t nfs4 -o nfsvers=4.2 -o actimeo=1 10.128.61.162:/longhornTest/ /var/lib/longhorn-backupstore-mounts/10_128_61_162/longhornTest], output mount.nfs4: Protocol not supported , error exit status 32: Cannot mount using NFSv4 , stderr, time="2022-12-27T08:12:37Z" level=error msg="Cannot mount nfs 10.128.61.162:/longhornTest/: vers=4.0: Failed to execute: mount [-t nfs4 -o nfsvers=4.0 -o actimeo=1 10.128.61.162:/longhornTest/ /var/lib/longhorn-backupstore-mounts/10_128_61_162/longhornTest], output mount.nfs4: Protocol not supported\n, error exit status 32: vers=4.1: Failed to execute: mount [-t nfs4 -o nfsvers=4.1 -o actimeo=1 10.128.61.162:/longhornTest/ /var/lib/longhorn-backupstore-mounts/10_128_61_162/longhornTest], output mount.nfs4: mounting 10.128.61.162:/longhornTest/ failed, reason given by server: No such file or directory\n, error exit status 32: vers=4.2: Failed to execute: mount [-t nfs4 -o nfsvers=4.2 -o actimeo=1 10.128.61.162:/longhornTest/ /var/lib/longhorn-backupstore-mounts/10_128_61_162/longhornTest], output mount.nfs4: Protocol not supported\n, error exit status 32: Cannot mount using NFSv4" , error exit status 1`

**Comment by derekbit** [comment]
Backup already supports NFSv4, NFSv4.1 and NFSv4.2. What's Longhorn version you're using? 
BTW, can you check if your Longhorn support NFSv4.1 (client)?

https://longhorn.io/docs/1.3.2/deploy/install/#installing-nfsv4-client

**Reply by FaraSys** [reply]
Longhorn version v1.3.2
root@atnode-01:/# cat /boot/config-`uname -r`| grep CONFIG_NFS_V4_1
CONFIG_NFS_V4_1=y
CONFIG_NFS_V4_1_IMPLEMENTATION_ID_DOMAIN="kernel.org"
CONFIG_NFS_V4_1_MIGRATION=y
root@atnode-01:/# cat /boot/config-`uname -r`| grep CONFIG_NFS_V4_2
CONFIG_NFS_V4_2=y
 '# CONFIG_NFS_V4_2_READ_PLUS is not set
CONFIG_NFS_V4_2_SSC_HELPER=y

**Comment by derekbit** [comment]
Just found `Failed to execute: mount [-t nfs4 -o nfsvers=4.1 -o actimeo=1 10.128.61.162:/longhornTest/ /var/lib/longhorn-backupstore-mounts/10_128_61_162/longhornTest], output mount.nfs4: mounting 10.128.61.162:/longhornTest/ failed, reason given by server: No such file or directory`. Looks you don't have the directory in the server side.

**Reply by FaraSys** [reply]
I feel so ashamed. Didn't notice this typo.
Fixed it and now it works.
Thank you!

**Reply by derekbit** [reply]
No worries. :)
